Sources/Internal/DoubleLayoutPassHelpers.swift

```diff
@@ -0,0 +1,136 @@
+
+extension CalendarView {
+
+  func installDoubleLayoutPassSizingLabel() {
+    doubleLayoutPassSizingLabel.removeFromSuperview()
+    addSubview(doubleLayoutPassSizingLabel)
+    subviews.first.map(doubleLayoutPassSizingLabel.sendSubviewToBack(_:))
+
+    doubleLayoutPassSizingLabel.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate([
+      doubleLayoutPassSizingLabel.leadingAnchor.constraint(
+        equalTo: layoutMarginsGuide.leadingAnchor),
+      doubleLayoutPassSizingLabel.trailingAnchor.constraint(
+        equalTo: layoutMarginsGuide.trailingAnchor),
+      doubleLayoutPassSizingLabel.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+      doubleLayoutPassSizingLabel.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+    ])
+  }
+
+  public override func invalidateIntrinsicContentSize() {
+    doubleLayoutPassSizingLabel.invalidateIntrinsicContentSize()
+  }
+
+  public override func setContentHuggingPriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentHuggingPriority(priority, for: axis)
+  }
+
+  public override func setContentCompressionResistancePriority(
+    _ priority: UILayoutPriority,
+    for axis: NSLayoutConstraint.Axis)
+  {
+    doubleLayoutPassSizingLabel.setContentCompressionResistancePriority(priority, for: axis)
+  }
+
+}
+
+// MARK: - WidthDependentIntrinsicContentHeightProviding
+
+protocol WidthDependentIntrinsicContentHeightProviding: CalendarView {
+
+  func intrinsicContentSize(forHorizontallyInsetWidth width: CGFloat) -> CGSize
+
+}
+
+// MARK: - DoubleLayoutPassSizingLabel
+
+final class DoubleLayoutPassSizingLabel: UILabel {
+
+  // MARK: Lifecycle
+
+  init(provider: WidthDependentIntrinsicContentHeightProviding) {
+    self.provider = provider
+
+    super.init(frame: .zero)
+
+    numberOfLines = 0
+    isUserInteractionEnabled = false
+    isAccessibilityElement = false
+  }
+
+  @available(*, unavailable)
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: Internal
+
+  override var intrinsicContentSize: CGSize {
+    guard let provider = provider else {
+      preconditionFailure(
+        "The sizing label's `provider` should not be `nil` for the duration of the its life")
+    }
+    if preferredMaxLayoutWidth == 0 {
+      return super.intrinsicContentSize
+    } else {
+      return provider.intrinsicContentSize(forHorizontallyInsetWidth: preferredMaxLayoutWidth)
+    }
+  }
+
+  override func draw(_: CGRect) {
+    // Do nothing, this view doesn't render any visual content.
+  }
+
+  override func drawText(in _: CGRect) {
+    // Do nothing, this view doesn't render any visual content.
+  }
+
+  // MARK: Private
+
+  private weak var provider: WidthDependentIntrinsicContentHeightProviding?
+
+}
+
```

Comment:
this whole file is just an adaptation of the one I created for the Airbnb app. I removed Airbnb-specific language and made this _less_ general since it's only used internally by `CalendarView`.